src/processors/controllers/processorController.js

```diff
@@ -54,3 +54,31 @@ module.exports.redownloadJSProcessor = async function (req, res, next) {
         next(e);
     }
 };
+
+module.exports.updateProcessor = async function (req, res, next) {
+    let { body: updatedProcessor, params: { processor_id: processorId }, query: { dl = false } } = req;
+    try {
+        dl = convertDlQueryParamToBoolean(dl);
+        const processor = await processorManager.updateProcessor(processorId, updatedProcessor, dl);
+        res.status(200).json(processor);
+    } catch (e) {
+        next(e);
+    }
+};
+
+function convertDlQueryParamToBoolean(dl) {
```

Comment:
lets convert this logic to the openapi3 spec

Reply:
the `express-swagger-validator` does not convert the string to boolean, did I miss anything in the spec?

Reply:
another example can be seen in GET /processors where I have used parseInt() in order to convert the `from` and `limit` parameters to proper integers even though the spec specifies that they are integers

Reply:
you're right